app/src/test/java/nu/yona/app/api/manager/impl/AuthenticateManagerImplTest.java

```diff
@@ -10,65 +10,230 @@
+public class AuthenticateManagerImplTest extends YonaTestCase {
     private AuthenticateManagerImpl manager;
+    private AuthenticateNetworkImpl authenticateNetworkImplMock;
+    private AuthenticateDAO authenticateDAOMock;
+    private RegisterUser registerUser;
+
+    private String correctMobileNumber = "+919686270640";
+    private String wrongMobileNumber = "+9999211";
+    private OTPVerficationCode correctOtpCode = new OTPVerficationCode("1234");
+    private OTPVerficationCode wrongOtpCode = new OTPVerficationCode("1111");
+    private String nullOtpCode = null;
+
+    private DataLoadListener genericResponseListener = new DataLoadListener() {
+        @Override
+        public void onDataLoad(Object result) {
+            assertTrue(result instanceof User);
+        }
+        @Override
+        public void onError(Object errorMessage) {
+            assertTrue(errorMessage instanceof  ErrorMessage);
+        }
+    };
+
 
     @Before
     public void setUp() throws Exception {
+        setUpApplicationTestData();
+        setUpRegisterUser();
         manager = new AuthenticateManagerImpl(YonaApplication.getAppContext());
+        mockRequiredClasses();
+        setUpMockedAuthNetworkDaoMethods();
+        setUpMockedAuthNetworkImplMethods();
     }
 
     @Test
-    public void checkRegisterUser() {
-        RegisterUser registerUser = new RegisterUser();
-        registerUser.setFirstName("Kinnar");
-        registerUser.setLastName("Vasa");
-        registerUser.setMobileNumber("+31873449748");
-        registerUser.setNickName("Kinnar");
-
-        manager.registerUser(registerUser, true, new DataLoadListener() {
-            @Override
-            public void onDataLoad(Object result) {
-                validateMobileNumber(password, ((User) result).getMobileNumberConfirmationCode());
-                assertTrue(result instanceof User);
-            }
+    public void verifyUserRegistrationWithProperData() {
+        registerUser.setMobileNumber(correctMobileNumber);
+        verifyUserRegistration();
+    }
 
-            @Override
-            public void onError(Object errorMessage) {
-                assertTrue(errorMessage instanceof ErrorMessage);
-            }
-        });
+    @Test
+    public void verifyUserRegistrationWithInvalidData() {
+        registerUser.setMobileNumber(wrongMobileNumber);
+        verifyUserRegistration();
+    }
+
+    @Test
+    public void verifyUserPassCodeWithCorrectCode() {
+        validateMobileNumber(correctOtpCode.getCode());
+    }
+
+    @Test
+    public void verifyUserPassCodeWithWrongCode() {
+        validateMobileNumber(wrongOtpCode.getCode());
+    }
+
+
+    @Test
+    public void verifyUserPassCodeWithNullCode() {
+        validateMobileNumber(nullOtpCode);
+    }
+
+    private void verifyUserRegistration(){
+        manager.registerUser(registerUser, true, genericResponseListener);
+    }
+
+
+    private void validateMobileNumber( String code) {
```

Comment:
This method mimics confirming the mobile number rather than validating it, so please rename into ``confirmMobileNumber``.

Reply:
Done.